python/tests/test_tables.py

```diff
@@ -1497,6 +1497,17 @@ def test_asdict(self):
         self.assertEqual(set(d1.keys()), set(d2.keys()))
         # TODO test the fromdict constructor
 
+    def test_iter(self):
+        def test_iter(table_collection):
+            table_names = [
+                attr_name for attr_name in sorted(dir(table_collection))
+                if isinstance(getattr(table_collection, attr_name), tskit.BaseTable)]
+            for n in table_names:
+                yield n, getattr(table_collection, n)
+        ts = msprime.simulate(10, mutation_rate=1, random_seed=1)
+        for t1, t2 in zip(test_iter(ts.tables), ts.tables):
```

Comment:
This won't spot if one iterator is longer than the other. Need to either make a list of it or use zip_longest.

Reply:
Well caught.